fix: prevent tooltip overflow in comment area

nehaaprasad opened the pull request:
### Problem
- Tooltip (mention list) overflows comment area boundaries when hovering over usernames, making it partially hidden and unusable.

#33509

### Solution
- Add viewport boundary checks to prevent tooltip from overflowing
- Increase z-index to ensure tooltip appears above other elements
- Fixes tooltip overflow issue when hovering over usernames in comments

### Changes
- Modified `quill.mention.js`: Added viewport boundary checks in `setMentionContainerPosition()`
- Modified `quill.scss`: Increased z-index from 1 to 9999 for `.ql-mention-list-container`

### Testing
- Tooltip now stays within viewport boundaries
- Tooltip appears above other elements
- No impact on existing functionality

nehaaprasad committed:
fix: prevent tooltip overflow in comment area

- Add viewport boundary checks to prevent tooltip from overflowing
- Increase z-index to ensure tooltip appears above other elements
- Fixes tooltip overflow issue when hovering over usernames in comments

iamejaaz commented:
Hello,
Since this is a UI-level change, could you please attach a video or GIF showing the before and after in the description?

nehaaprasad commented:
@iamejaaz 
This one also tested!

iamejaaz commented:
> @iamejaaz This one also tested!

<img width="1530" height="754" alt="image" src="https://github.com/user-attachments/assets/4c50d5fe-4b3e-49d1-835d-4958caf18a74" />

Hello @naaa760 ,
I tried this, but this PR doesn’t fix the issue.

nehaaprasad commented:
@iamejaaz 
Okay, I will fix the issue. Thanks for letting me know

nehaaprasad committed:
fix: improve tooltip overflow prevention in comment areas

- Change tooltip positioning from absolute to fixed positioning
- Append tooltip to document.body to prevent container clipping
- Add comprehensive viewport boundary checks for all edges
- Implement intelligent positioning (below first, above if needed)
- Add scroll and resize event handlers for dynamic repositioning
- Increase z-index to 999999 to ensure tooltip visibility
- Add proper cleanup method to prevent memory leaks

Fixes tooltip overflow when hovering over usernames in comments
that are positioned near viewport edges.

nehaaprasad commented:
@iamejaaz 
Please check again, is it fixed now?

iamejaaz commented:
> @iamejaaz Please check again, is it fixed now?

It still isn't fixed. Could you please check on your end first? Also, please attach screenshots, as this PR includes UI-level changes.
